lib/hammox.ex

```diff
@@ -109,10 +109,25 @@ defmodule Hammox do
 
   add_2.(1.5, 2.5) # throws Hammox.TypeMatchError
   ```
+
+  If pass module and list instead of MFA and module works like `protect/3`
+  ```elixir
+  # calling this
+  Hammox.protect(SomeModule, foo: 1)
+  # works same as this
+  Hammox.protect(SomeModule, SomeModule, foo: 1)
+  ```
   """
-  @spec protect(mfa :: mfa(), behaviour_name :: module()) :: fun()
+  @spec protect(
+          mfa() | module(),
+          module() | [{atom(), arity() | [arity()]}]
+        ) ::
+          fun() | map()
   def protect(mfa, behaviour_name)
 
+  def protect(module_name, funs) when is_atom(module_name and is_list(funs)),
+    do: protect(module_name, module_name, funs)
+
   def protect({module_name, function_name, arity}, behaviour_name)
```

Comment:
I'd split this typespec into multiple typespecs to signify these are separate call signatures:
```suggestion
  def protect(mfa, behaviour_name)

  @spec protect(module_name :: module(), funs :: [{atom(), arity() | [arity()]}]) :: map()
  def protect(module_name, funs) when is_atom(module_name and is_list(funs)),
    do: protect(module_name, module_name, funs)

  @spec protect(mfa :: mfa(), behaviour_name :: module()) :: fun()
  def protect({module_name, function_name, arity}, behaviour_name)
```